### PR TITLE
[kube-proxy] Fix Jump() bug in pkg/proxy/iptables/proxier_test.go

### LINKED ISSUE #36043: [Kube-proxy] hasJump() in proxier_test.go has bug?
**Is this a BUG REPORT?**

Yes

**What?**

It seems that [hasJump()](https://github.com/kubernetes/kubernetes/blob/master/pkg/proxy/iptables/proxier_test.go#L507) in proxier_test.go do some wrong matches. It failed some test cases I designed.

1.  

```
rules : []iptablestest.Rule{
	    iptablestest.Rule{"--dport ":"3001", "-p ":"tcp", "-j ":"REJECT"},
	},
	destChain: "REJECT",
	destIP: "1.2.3.4",
	destPort: "3001",
```

expect **true**, but got **false**.

but, 

```
rules : []iptablestest.Rule{
	    iptablestest.Rule{"--dport ":"3001", "-p ":"tcp", "-j ":"REJECT"},
	},
	destChain: "REJECT",
	destIP: "",
	destPort: "3001",
```

will got **true** as expected.

2. 

```
rules : []iptablestest.Rule{
	    iptablestest.Rule{"-d ":"10.20.30.41/32", "--dport ":"80", "-p ":"tcp", "-j ":"REJECT"},
	},
	destChain: "REJECT",
	destIP: "10.20.30.41",
	destPort: "8081",
```

expect **false**, but got **true**.

3. immediately return **false** when first rule fail to match even though second rule will **match**

```
rules : []iptablestest.Rule{
	    iptablestest.Rule{"-d ":"10.20.30.41/32", "--dport ":"80", "-p ":"tcp", "-j ":"REJECT"},
            iptablestest.Rule{"-d ":"1.2.3.4/32", -p ":"tcp", "-j ":"REJECT"},
	},
	destChain: "REJECT",
	destIP: "1.2.3.4",
        destPort: "",
```

expect **true**, but got **false**.

@thockin @bprashanth 

Am I right? If so, I will send a PR to fix it.

Thanks!

### PR DESCRIPTION
**Pull request by m1093782566**
**Which issue this PR fixes**

Fixes #36043

---
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/36055)

### CODE REVIEW

**Comment by thockin** [comment]
ok to test

File: pkg/proxy/iptables/proxier_test.go [hunk 1]
```diff
@@ -505,18 +506,129 @@ func NewFakeProxier(ipt utiliptables.Interface) *Proxier {
...
+			expected:  false,
+		},
+		{ // case 4
+			rules: []iptablestest.Rule{
+				iptablestest.Rule{"-d ": "1.2.3.4/32", "--dport ": "80", "-p ": "tcp", "-j ": "KUBE-XLB-GF53O3C2HZEXL2XN"},
+			},
+			destChain: "KUBE-XLB-GF53O3C2HZEXL2XN",
+			destIP:    "10.20.30.40",
+			destPort:  "80",
+			expected:  false,
```
Comment (thockin) [inline pkg/proxy/iptables/proxier_test.go]: same

File: pkg/proxy/iptables/proxier_test.go [hunk 2]
```diff
@@ -505,18 +506,129 @@ func NewFakeProxier(ipt utiliptables.Interface) *Proxier {
...
+			expected:  true,
+		},
+		{ // case 3
+			rules: []iptablestest.Rule{
+				iptablestest.Rule{"-d ": "1.2.3.4/32", "--dport ": "80", "-p ": "tcp", "-j ": "KUBE-XLB-GF53O3C2HZEXL2XN"},
+			},
+			destChain: "KUBE-XLB-GF53O3C2HZEXL2XN",
+			destIP:    "1.2.3.4",
+			destPort:  "8080",
+			expected:  false,
```
Comment (m1093782566) [inline pkg/proxy/iptables/proxier_test.go]: Okay

File: pkg/proxy/iptables/proxier_test.go (hunk 1 above)
Comment (m1093782566) [inline pkg/proxy/iptables/proxier_test.go]: Okay

**Comment by m1093782566** [comment]
@thockin 

Comments are fixed and CI is paased, PTAL.

**Pull request merged**